lib/pbench/test/unit/server/test_schema.py

```diff
@@ -122,18 +122,17 @@ def test_invalid_required(self, test):
         assert x.invalid(json) is expected
 
     @pytest.mark.parametrize(
-        "test",
-        (
-            ({"data": "yes"}, False),
-            ({"data": None}, True),
-            ({"foo": "yes"}, False),
-            ({"foo": None}, False),
-        ),
+        "test", ({"data": "yes"}, {"data": None}, {"foo": "yes"}, {"foo": None},),
     )
     def test_invalid_optional(self, test):
+        """
+        An optional parameter is either present or not: either is OK, and the
+        value None is acceptable. (In other words, the "invalid" test isn't
+        meaningful for required=False parameters, and should always succeed.)
+        """
         x = Parameter("data", ParamType.STRING, required=False)
-        json, expected = test
-        assert x.invalid(json) is expected
+        json = test
+        assert x.invalid(json) is False
```

Comment:
[PEP8](https://www.python.org/dev/peps/pep-0008/#id51) recommends against using `is` with boolean values.
```
        assert not x.invalid(json)
```